Added aws-request-signing. (#5131)
Kong/docs.konghq.com: app/_hub/TheLEGOGroup/aws-request-signing/_index.md

---
# This file is for documenting an individual Kong plugin.
#
# 1. Duplicate this file in your own *publisher path* on your own branch.
# Your publisher path is relative to app/_hub/.
# The path must consist only of alphanumeric characters and hyphens (-).
#
# 2. Duplicate the versions.yml file into your new plugin directory.
# Set the Kong Gateway version that the plugin is being added to.

# 3. Add a 64x64px icon for the plugin to app/_assets/images/icons/hub.
# The name of the file must be in the following format: <publisher>_<plugin-directory-name>.png
# For example, for the rate limiting plugin the icon name is kong-inc_rate-limiting.png
# If your plugin doesn't have an icon yet, you can duplicate the default_icon.png file.
#
# 4. Fill in the template in this file.
#
# The following YAML data must be filled out as prescribed by the comments
# on individual parameters. Also see documentation at:
# https://github.com/Kong/docs.konghq.com/app/_hub for examples.
# Remove inapplicable entries and comments as needed.

> name: aws-request-signing 

Reviewer: ```suggestion
name: AWS Request Signing
```

  # If this is being published and supported by a company, please use your company name.
  # Note that every extension by a given publisher must have the exact same value.

categories: 
  # (required) Uncomment ONE that applies.
  #- authentication
  #- security
  #- traffic-control
  - serverless
  #- analytics-monitoring
  #- transformations
  #- logging
  #- deployment
# Array format only; if your plugin applies to multiple categories,
# uncomment the most applicable category.

type: plugin 
  # (required) String, one of:
  # plugin          | extensions of the core platform
  # integration     | extensions of the Kong Admin API

desc: Sign requests with AWS SIGV4 and temp credentials for secure use of AWS Lambdas in Kong. # (required) 1-liner description; max 80 chars
description: |
  This plugin allows for secure communication with AWS Lambdas. It signs requests with AWS SIGV4 and temporary credentials obtained from sts.amazonaws.com using an OAuth token. 
  This eliminates the need for an AWS API Gateway and simplifies the use of Lambdas as upstreams in Kong. 
  
  However, in order to use this plugin, there is an AWS setup required.

>   Specifically, you will need to add your token issuer to the "Identity Providers" in your AWS account, this way the plugin can request temporary credentials. More details can be seen in the [plugin repo.](https://github.com/LEGO/kong-aws-request-signing#aws-setup-required)

Reviewer 2: ```suggestion
  Specifically, you will need to add your token issuer to the "Identity Providers" in your AWS account, this way the plugin can request temporary credentials. 
 
 For more information on the required AWS setup, visit the [plugin repo.](https://github.com/LEGO/kong-aws-request-signing#aws-setup-required)
```

categories: 
  # (required) Uncomment ONE that applies.
  #- authentication
  #- security
  #- traffic-control
  - serverless
  #- analytics-monitoring
  #- transformations
  #- logging
  #- deployment
# Array format only; if your plugin applies to multiple categories,
# uncomment the most applicable category.

type: plugin 
  # (required) String, one of:
  # plugin          | extensions of the core platform
  # integration     | extensions of the Kong Admin API

desc: Sign requests with AWS SIGV4 and temp credentials for secure use of AWS Lambdas in Kong. # (required) 1-liner description; max 80 chars
description: |
  This plugin allows for secure communication with AWS Lambdas. It signs requests with AWS SIGV4 and temporary credentials obtained from sts.amazonaws.com using an OAuth token. 
  This eliminates the need for an AWS API Gateway and simplifies the use of Lambdas as upstreams in Kong. 
  
  However, in order to use this plugin, there is an AWS setup required.
  Specifically, you will need to add your token issuer to the "Identity Providers" in your AWS account, this way the plugin can request temporary credentials. 
 

>  For more information on the required AWS setup, visit the [plugin repo.](https://github.com/LEGO/kong-aws-request-signing#aws-setup-required)

Reviewer: ```suggestion
  For more information on the required AWS setup, visit the [plugin repo.](https://github.com/LEGO/kong-aws-request-signing#aws-setup-required)
```
